dev/set_matrix.py

```diff
@@ -316,8 +316,7 @@ def process_requirements(requirements, version=None):
             )
             if match_all:
                 return packages
-        else:
-            return []
+        return []
```

Comment:
Required this change because `pylint` now checks scripts in the `dev` directory.

Reply:
So pylint print warning for `for ... else` usage ?

Reply:
No, pylint raises a warning when we use `for-else` without a `break` statement in the loop:

https://docs.quantifiedcode.com/python-anti-patterns/correctness/else_clause_on_loop_without_a_break_statement.html

```python
# Example

def contains_magic_number(list, magic_number):
    for i in list:
        if i == magic_number:
            print("This list contains the magic number")
    else:
        print("This list does NOT contain the magic number")
```